Python/challenges/day_7.py

```diff
@@ -0,0 +1,72 @@
+def find_smallest_dir_with_update_size(tree, update_size):
+    smallest = None
+    current_space = tree["/"]
+    space_to_free = abs(70000000 - (current_space + update_size))
+    for value in tree.values():
+        if value > space_to_free:
+            if smallest == None :
+                smallest = value
+            if smallest > value:
+                smallest = value
+    return smallest
+
+def get_sum_dirs_under_max_size(tree, max_size):
+    sum = 0
+    for value in tree.values():
+        if value < max_size:
+            sum += value
+    return sum
+
+def count_directory_size(input):
+    sum = 0
+    line = input[0].split(" ")
+    while line[0] != "$":
+        line = input.pop(0).split(" ")
+        if line[0] != "dir":
+            sum += int(line[0])
+        if len(input) == 0:
+            break
+        line = input[0].split(" ")
+    return input, sum
+    
+def go_directory_back(directory):
+    new_directory = ""
+    directory = directory.split("/")
+    for i in range(len(directory) - 2):
+        new_directory += directory[i] + "/"
+    return new_directory
```

Comment:
Good idea to use a method like this. However, can you think of a solution that doesn't require a loop?